src/app/productConfig/productConfig.component.js

```diff
@@ -0,0 +1,90 @@
+import angular from 'angular';
+
+import template from './productConfig.tpl';
+import templateModal from './productConfigModal.tpl';
+import designationsService from 'common/services/api/designations.service';
+
+let componentName = 'productConfig';
+
+class ProductConfigController{
+
+  /* @ngInject */
+  constructor($uibModal) {
+    this.$uibModal = $uibModal;
+  }
+
+  configModal(){
+    var productCode = this.productCode;
+
+    this.$uibModal.open({
+      templateUrl: templateModal.name,
+      controller: ModalInstanceCtrl,
+      controllerAs: '$ctrl',
+      size: 'lg give-modal',
+      resolve: {
+        productData: [designationsService.name, function(designationsService){
+          return designationsService.productLookup(productCode);
```

Comment:
@OzzieOrca How can I convert this to a promise so it will work as a resolve should?